include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -164,19 +188,24 @@ namespace internal
 
                   for (unsigned int j = 0; j < this->finite_elements[i]->size();
                        ++j)
-                    if ((*this->finite_elements[i])[j].reference_cell_type() !=
+                    if ((*this->finite_elements[i])[j].reference_cell_type() ==
                         ReferenceCell::get_hypercube(dim))
-                      quadrature.push_back(*quadrature1);
-                    else
                       quadrature.push_back(*quadrature2);
+                    else if ((*this->finite_elements[i])[j]
+                               .reference_cell_type() ==
+                             ReferenceCell::Type::Wedge)
+                      quadrature.push_back(*quadrature3);
+                    else
+                      quadrature.push_back(*quadrature1);
```

Comment:
You'll get into trouble here if another cell type appears (namely, once you have pyramids). You'll have to *know* that you need to add something new here. Better to use a `switch(....reference_cell_type())` statement that's explicit about what cases each of these corresponds to, with an explicit `Assert(false,...)` in the `default:` branch.